src/provider/common/Injector.ts

```diff
@@ -82,22 +82,34 @@ export class Injector {
                 container.bind(Inject[key]).toConstantValue(Bindings[key]);
             }
         });
+
+        Injector._initialized = Promise.all(promises).then(() => {});
+        return container;
+    })();
+
+    public static get initialized(): Promise<void> {
+        return this._initialized;
+    }
+
+    public static async init(): Promise<void> {
+        const container: Container = Injector._container;
+        const promises: Promise<unknown>[] = [];
+
         Object.keys(Bindings).forEach(k => {
             const key: Keys = k as any;
             const proto = (Bindings[key] as Function).prototype;
 
             if (proto && proto.hasOwnProperty('init')) {
                 const instance = (container.get(Inject[key]) as AsyncInit);
-                promises.push(instance.delayedInit());
+                if (instance.delayedInit) {
+                    promises.push(instance.delayedInit());
+                }
             }
         });
 
-        Injector._initialized = Promise.all(promises).then(() => {});
-        return container;
-    })();
+        Injector._initialized = Promise.all(promises).then();
```

Comment:
Just adding that the FDC3 Injector has changed in my PR https://github.com/HadoukenIO/fdc3-service/pull/134/files.  It might be worth just copying that whole file over